### PR TITLE
Value {map, relation} editor widget: fix representation problem with < character

### PR DESCRIPTION
**Pull request by m-kuhn**
Fixes https://github.com/opengisch/QField/issues/5489

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1892fb65cf95ff9da7f5cf7e2668572900a6de75: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5543-e0f5fbcc849b36ac0e7f00639997ea0a7f9061ea-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5543-e0f5fbcc849b36ac0e7f00639997ea0a7f9061ea-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5543-e0f5fbcc849b36ac0e7f00639997ea0a7f9061ea-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5543-e0f5fbcc849b36ac0e7f00639997ea0a7f9061ea-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5543-e0f5fbcc849b36ac0e7f00639997ea0a7f9061ea-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5543-e0f5fbcc849b36ac0e7f00639997ea0a7f9061ea-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5543-e0f5fbcc849b36ac0e7f00639997ea0a7f9061ea-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5543-e0f5fbcc849b36ac0e7f00639997ea0a7f9061ea-android-x86.apk)
</details>

**Comment by nirvn** [comment]
@m-kuhn , I had a similar fix in mind however this will create a problem when highlighting search matches. Ie, if you search for "amp", the simple highlighter will match the amp part of &amp;.

**Comment by nirvn** [comment]
We'd need a regex logic that excludes matches between & and ; for it to work well

File: src/qml/RelationCombobox.qml [hunk 1]
```diff
@@ -184,9 +184,10 @@ Item {
           id: delegateRect
 
           property int idx: index
-          property string itemText: featureListModel.searchTerm != '' ? displayString.replace(new RegExp('(' + featureListModel.searchTerm + ')', "i"), '<span style="text-decoration:underline;' + Theme.toInlineStyles({
+          readonly property string encodedDisplayString = displayString.replaceAll('&', '&amp;').replaceAll('<', '&lt;').replaceAll('>', '&gt;').replaceAll('"', '&quot;').replaceAll("'", '&#039;')
+          property string itemText: featureListModel.searchTerm != '' ? encodedDisplayString.replace(new RegExp('(' + featureListModel.searchTerm + ')', "i"), '<span style="text-decoration:underline;' + Theme.toInlineStyles({
```
Comment (nirvn) [inline src/qml/RelationCombobox.qml:188]: This regexp should work _if_ the replace allows for long-ahead and look-behind:

```suggestion
          property string itemText: featureListModel.searchTerm != '' ? encodedDisplayString.replace(new RegExp('(?!=&[a-z]*)(' + featureListModel.searchTerm + ')(?![a-z]*;)', "i"), '<span style="text-decoration:underline;' + Theme.toInlineStyles({
```

Proof of logic:

image

**Comment by nirvn** [comment]
Similar fix needed here: https://github.com/opengisch/QField/blob/master/src/qml/editorwidgets/ValueMap.qml#L374

**Comment by nirvn** [comment]
@m-kuhn , I've pushed an updated commit which fixes both the relation combobox - used by value relation - as well as the value map. Tested locally, working great.

**Review by nirvn — APPROVED** [review]
@m-kuhn , I've approved due to you opening the PR. Feel free to merge when you're happy :)

**Comment by qfield-fairy** [comment]
The backport to `release-3_3` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply 13302f455... Value {map,relation} editor widgets: fix representation problem with < character
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".
hint: Disable this message with "git config advice.mergeConflict false"

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging src/qml/RelationCombobox.qml
CONFLICT (content): Merge conflict in src/qml/RelationCombobox.qml
Auto-merging src/qml/editorwidgets/ValueMap.qml
CONFLICT (content): Merge conflict in src/qml/editorwidgets/ValueMap.qml

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-release-3_3 release-3_3
# Navigate to the new working tree
cd .worktrees/backport-release-3_3
# Create a new branch
git switch --create backport-5543-to-release-3_3
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 13302f45546908bc34790b91c91831b8cda94edb,1ad84b9d9a2ce99c487ee8698f1ad1259a24c49f,91c88fd609a4169118e33b3333854829bfc075e1,8785fbd17614609d7042244b543ce8d029d688d9
# Push it to GitHub
git push --set-upstream origin backport-5543-to-release-3_3
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-release-3_3
```
Then, create a pull request where the `base` branch is `release-3_3` and the `compare`/`head` branch is `backport-5543-to-release-3_3`.

**Pull request merged**